processor/lsmintervalprocessor/processor.go

```diff
@@ -39,8 +39,38 @@ import (
 var _ processor.Metrics = (*Processor)(nil)
 var zeroTime = time.Unix(0, 0).UTC()
 
-// TODO (lahsivjar): Optimize pebble
-const batchCommitThreshold = 16 << 20 // 16MB
+const (
+	// pebbleMemTableSize defines the max steady state size of a memtable.
+	// There can be more than 1 memtable in memory at a time as it takes
+	// time for old memtable to flush. The memtable size also defines
+	// the size for large batches. A large batch is a batch which will
+	// take atleast half of the memtable size. Note that the Batch#Len
+	// is not the same as the memtable size that the batch will occupy
+	// as data in batches are encoded differently. In general, the
+	// memtable size of the batch will be higher than the length of the
+	// batch data.
+	//
+	// On commit, data in the large batch maybe kept by pebble and thus
+	// large batches will need to be reallocated. Note that large batch
+	// classification uses the memtable size that a batch will occupy
+	// rather than the length of data slice backing the batch.
+	pebbleMemTableSize = 64 << 20 // 64MB
```

Comment:
[**For reviewers**] I have gone a bit high here, this would mean we would require 64*2=128MB of memory at-least (due to max number of memtables as configured via `pebbleMemTableStopWritesThreshold`) for memtable. This should be good for most of our use cases - also, we could expose these configs in the component configuration for tweaking in future but I haven't done that so far.